drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,269 @@
+
+    if (tx.nVersion >= 2) {
+        for (int js = 0; js < joinsplits; js++) {
+            ...
+        }
+    }
+
+When (de)serializing v3 transactions, the version group id should also be checked in case the transaction is intended for a branch which has a different format for its version 3 transaction::
+
+    if (tx.nVersion == 3 && tx.nVersionGroupID == OVERWINTER_VERSION_GROUP_ID) {
+        auto expiryHeight = tx.nExpiryHeight;
+    }
+
+Tests can continue to set the version to zero as an error condition::
+
+    mtx.nVersion = 0
+
+
+Overwinter Validation
+---------------------
+
+To test if the format of an Overwinter transaction is v3 or not::
+
+    if (tx.fOverwintered && tx.nVersion == 3) {
+        // Valid v3 format transaction
+    }
+
+This only tests that the format of the transaction matches the v3 specification described above.
+
+To test if the format of an Overwinter transaction is bothv3 and the transaction itself is intended for the client's chain::
+
+    if (tx.fOverwintered &&
+        tx.nVersionGroupID == OVERWINTER_VERSION_GROUP_ID) &&
+        tx.nVersion == 3) {
+        // Valid v3 format transaction intended for this client's chain
+    }
+
+However, it's possible that a ZClone is using the same version group id and passes the conditional.
+
+Ultimately, a client can determine if a transaction is truly intended for the client's chain or not by following the signature verification process detailed in the Transaction Signature Verification for Overwinter ZIP [#zip-???]_.
+
+Deployment
+==========
+
+This proposal will be deployed with the Overwinter network upgrade.
+
+Testnet is set to activate Overwinter at block XXX.
+
+- This means that starting from block XXX of testnet, new Overwinter consensus rules take effect and transactions must be using v3 to be accepted as valid.
+
+Mainnet is set to activate Overwinter at block XXX.
+
+- This means that starting from block XXX of mainnet, new Overwinter consensus rules take effect and transactions must be using v3 to be accepted as valid.
+
+
+Backwards compatibility
+=======================
+
+This proposal intentionally creates what is known as a "bilateral hard fork" between pre-Overwinter software and Overwinter compatible software. Use of this new transaction format requires that all network participants upgrade their software to a compatible version within the upgrade window. Pre-Overwinter software will treat Overwinter transactions as invalid.  Overwinter compatible software will reject legacy transactions.  Once Overwinter has activated, nodes will only accept transactions based upon supported transaction version numbers and recognized version group ids.
+
+
+Reference Implementation
+========================
+
+https://github.com/zcash/zcash/pull/2925
+
+References
+==========
+
+Design hard fork activation mechanism https://github.com/zcash/zcash/issues/2286
```

Comment:
Either add a reference to this from the body of the document (in Abstract or Motivation), or delete it.